Assets/Python/OOSLogger.py

```
@@ -1,190 +1,462 @@
+            # Unit info section
+            w("\n\nUnit Info:\n----------\n")
+            if pPlayer.getNumUnits():
+                for pUnit in pPlayer.units():
+                    w("Player %d, Unit ID: %d, %s\n" % (iPlayer, pUnit.getID(),
+                                                                    TextUtil.convertToStr(pUnit.getName())))
+                    w("X: %d, Y: %d\nDamage: %d\n" % (pUnit.getX(), pUnit.getY(), pUnit.getDamage()))
+                    w("Experience: %d\nLevel: %d\n" % (pUnit.getExperience(), pUnit.getLevel()))
+
+                    # Collect promotions
+                    wrote_header = False
+                    for j in xrange(GC.getNumPromotionInfos()):
```

Ruff (0.12.2)

Undefined name `xrange`

(F821)